spark3/src/main/java/org/apache/iceberg/spark/SparkSessionCatalog.java

```diff
@@ -280,4 +282,10 @@ private boolean useIceberg(String provider) {
 
     return false;
   }
+
+  private T getSessionCatalog() {
+    Preconditions.checkNotNull(sessionCatalog, "Delegated SessionCatalog is not provided by Spark. " +
+        "Please make sure you are replacing Spark's default catalog.");
```

Comment:
How about adding the name of the session catalog, like this: "Please make sure your are replacing Spark's default catalog, named 'spark_catalog'".

Also, the first line makes it sound like Spark did something wrong when this is probably a misconfiguration. For that, what about changing it to "Delegated SessionCatalog is missing" rather that "not provided by Spark"?

Reply:
Ah right. I agree that's not something Spark is doing wrong and the sentence may leads misunderstanding. Thanks for the suggestion!